components/store/FulfillOrderDialog.js

```diff
@@ -0,0 +1,184 @@
+// Module imports
+import React from 'react'
+import PropTypes from 'prop-types'
+
+
+
+
+// Component imports
+import { connect } from '../../store'
+import Dialog from '../Dialog'
+import Component from '../Component'
+import ValidatedFormInput from '../ValidatedFormInput'
+
+
+
+
+
+// Component constants
+const INVALID_TRACKING_NUMBER = 'Tracking Number is Required'
+
+
+
+@connect
+class CartUpdateDialog extends Component {
+  /***************************************************************************\
+    Class Properties
+  \***************************************************************************/
+
+  state = {
+    trackingNumber: '',
+    submitting: false,
+  }
+
+
+
+
+
+  /***************************************************************************\
+    Private Methods
+  \***************************************************************************/
+
+  _handleSubmit = async () => {
+    const {
+      onClose,
+      orderId,
+    } = this.props
+
+    const {
+      trackingNumber,
+    } = this.state
+
+    this.setState({ submitting: true })
+
+    const { status } = await this.props.updateOrder(orderId, {
+      status: 'fulfilled',
+      shipping: {
+        carrier: 'Royal Mail',
+        tracking_number: trackingNumber,
+      },
+    })
+
+    if (status === 'success') {
+      onClose()
+    } else {
+      this.setState({
+        error: 'Looks like something went wrong. Contact the techs',
```

Comment:
Clarify the process of contacting techrats. Expect this can be accessed by externals who have no idea how to do it, and provide either helpdesk link or an email.

Reply:
This is not a publicly available dialog. This is for fulfilling orders on the quartermaster side.